pallets/substrate-app/src/lib.rs

```diff
@@ -439,6 +447,35 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::call_index(8)]
+        #[pallet::weight(<T as Config>::WeightInfo::register_erc20_asset())]
+        pub fn update_transaction_status(
+            origin: OriginFor<T>,
+            message_id: H256,
+            transfer_status: XCMAppTransferStatus,
+        ) -> DispatchResult {
+            let CallOriginOutput {
+                network_id,
+                timepoint,
+                ..
+            } = T::CallOrigin::ensure_origin(origin)?;
+
+            let message_status = match transfer_status {
+                XCMAppTransferStatus::Success => {
+                    Self::deposit_event(Event::Done(message_id));
```

Comment:
bridge-proxy pallet emits event with message_id and message_status, so I'm not sure if this event is needed

Reply:
removed